### DISCUSSION TITLE
Physical size of domain

### DISCUSSION THREAD

**Discussion by RasoulNajafi** [neperfepx/neper#146] [Q&A] [answered]
Hi all,

I am new to use Neper. Maybe this question has a simple answer. I would appreciate if you help me to find an answer for it. 
I use the following command in order to create RVE:

neper -T -n 350 -id 1 -domain ''cube(1,1,1)'' -periodicity all -morpho '' diameq:lognormal(1,0.1),aspratio(2.2,1,1)''

As you can see I have to normalize the dimension of the real RVE into a cube with edge length of 1. In my project, it is quite necessary to have a physical dimension of the generated RVE as well. I tried to use statistics in order to obtain the equivalent diameter of the RVE located on the edge. I use the following code in tesselation module:

'' -statcell id,diameq'' 

The problem is, the values I obtain is also normalized. I also used the code above without ''cube(1,1,1)'' command and the values were also normalized and I didnt have the real grains equivalent diameter. 

My question is how I can extract the physical dimension of the RVE based on the input data. Is there anyway to directly measure the real length of the RVE edge? and how I can find the id number of the elements located on the edge?

Many thanks in advance 
Rasoul

**Comment by rquey** [comment, marked as answer]
If you provide the domain size and the number of grains, then the average grain size (the average value of the `lognormal`) will be interpreted as a _relative_ value.

If you provide the domain size and `-n from_morpho` for the number of grains, then the average grain size (the average value of the `lognormal`) will be interpreted as an _absolute_ value, and the number of grains will be determined accordingly.

There is no equivalent of `from_morpho` for the domain, but in your case, you typically have to provide absolute values to `-morpho` (as in `morpho "diameq:lognormal(0.01,0.001),..."`), `-n from_morpho`, and adjust the domain size (as in `-domain "cube(0.1,0.1,0.1)"`).
